core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb_test_export.go

```diff
@@ -74,7 +79,7 @@ func (env *testVDBEnv) stopExternalResource() {
 
 func (env *testVDBEnv) closeAndReopen() {
 	env.DBProvider.Close()
-	dbProvider, _ := NewVersionedDBProvider(env.config, &disabled.Provider{}, &statedb.Cache{})
+	dbProvider, _ := NewVersionedDBProvider(env.config, &disabled.Provider{}, env.sysNamespaces)
```

Comment:
In "env" you had obtained a pointer to cache. You would need that to point to new cache object here. Perhaps, no caching based test test is calling this function yet and hence escaped.

Reply:
I don't understand. Earlier, the `privacyenabledstate` was creating the cache. As we have moved to `statecouchdb` package, now we create a `cache` in`NewVersionedDBProvider()`. When we `closeAndReopen()`, we need to create a new cache object which is anyway done by this line. If I am missing something, I am not sure what you meant here.

Reply:
Look at the `env.init()` function. You have this line `env.cache = dbProvider.cache`. 
After you invoke this function (`closeAndReopen`) in some test, `env.cache` would still be pointing to older cache object, not the new one that will be created.

Reply:
Ah :-) Probably, we should have named `closeAndReopen()` as `reInit()`. I am saying that because it would have made me look at `init()` once after changing this code. 

Done.